src/textual/css/_color.py

```diff
@@ -0,0 +1,57 @@
+"""
```

Comment:
We're doing some custom parsing here, which is different from the parsing done in the CSS. I'd like to avoid solving the same problem twice.

We could use `tokenize_value` in textual.css.tokenize for this -- which is what the CSS parser uses.

Here's an example:

```python
>>> from textual.css.tokenize import tokenize_value
>>> list(tokenize_value("hsl( 180, 50%, 50% ) 30%", "expression"))
[Token(name='color', value='hsl( 180, 50%, 50% )', path='expression', code='hsl( 180, 50%, 50% ) 30%', location=(0, 0)), Token(name='whitespace', value=' ', path='expression', code='hsl( 180, 50%, 50% ) 30%', location=(0, 20)), Token(name='scalar', value='30%', path='expression', code='hsl( 180, 50%, 50% ) 30%', location=(0, 21))]
```

This should give you the same tokens passed to the builder object.

How about the property uses `tokenize` and calls a method to extract a Color from the tokens. The builder could call this same method.

Reply:
My reasoning was that simply doing a `join` over the tokens in the StylesBuilder was much faster than tokenizing the values in the ColorProperty - and the 2 regexes used to find the percentage values likely have a fast lookup since they only search at the beginning and the end of a string.
But ok, I'll do that :slightly_smiling_face: